lib/streamlit/elements/widgets/button.py

```diff
@@ -682,7 +682,7 @@ def _page_link(
                 if label is None:
                     page_link_proto.label = page_name
                 page_link_proto.page_script_hash = page_data["page_script_hash"]
-                page_link_proto.page = full_path
```

Comment:
page is fed to the `href` attribute in `PageLink.tsx` - this is what shows in the browser preview in the bottom left corner.